streams/src/test/java/org/apache/kafka/streams/kstream/internals/CogroupedKStreamImplTest.java

```diff
@@ -0,0 +1,127 @@
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.common.errors.InvalidTopicException;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.streams.kstream.CogroupedKStream;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.KStreamBuilder;
+import org.apache.kafka.streams.kstream.Merger;
+import org.apache.kafka.streams.kstream.SessionWindows;
+import org.apache.kafka.streams.kstream.TimeWindows;
+import org.apache.kafka.test.KStreamTestDriver;
+import org.apache.kafka.test.MockAggregator;
+import org.apache.kafka.test.MockInitializer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class CogroupedKStreamImplTest {
```

Comment:
Would it be possible to add some unit tests for the functionality, too? I know we have the integration tests, but it is always easier to debug problems in a unit test. In theory you should be able to use the `KStreamTestDriver` to test this